### PR TITLE
Extend firewall customizations to add sources

### PR DESCRIPTION
**Pull request by runcom**
~~Requires https://github.com/osbuild/osbuild/pull/1137~~ (see below)

This patch adds support to specify firewalld sources for zones

(will add tests asap)

Signed-off-by: Antonio Murdaca <runcom@linux.com>

---
osbuild-composer counterpart to osbuild/osbuild#1157.

Signed-off-by: Irene Diez <idiez@redhat.com>

This pull request includes:

- [x] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Comment by 7flying** [comment]
The tests are all green but the CentOS9 ones on Packit

**Comment by achilleas-k** [comment]
Looks good, but the git history is a bit weird with the addition of the customization and then changing it.  Not a big issue though.  Did a rebase to hopefully get everything green again and merge.

**Review by achilleas-k — DISMISSED** [review]

**Comment by 7flying** [comment]
> Looks good, but the git history is a bit weird with the addition of the customization and then changing it. Not a big issue though. Did a rebase to hopefully get everything green again and merge.

Yeah, I wanted to build upon @runcom's work now that I have permissions to write on other people's branches. Seemed more 'fair/accountable' to me.

Thanks for the review Achilleas!

**Comment by gicmo** [comment]
> > Looks good, but the git history is a bit weird with the addition of the customization and then changing it. Not a big issue though. Did a rebase to hopefully get everything green again and merge.
> 
> Yeah, I wanted to build upon @runcom's work now that I have permissions to write on other people's branches. Seemed more 'fair/accountable' to me.

We sometimes use the `Co-authored-By:`  git trailer for this, see https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

**Comment by mcattamoredhat** [comment]
Hello, I have been trying to test locally the firewall zones customization, may you please help me to clarify which is the proper syntax to include it in the blueprint?

Example 1: 
[customizations.firewall.zones]
name = "dmz"
sources = ["192.168.100.50"]

or

Example 2:
[[customizations.firewall.zones]]
name = "dmz"
sources = ["192.168.100.50"]

Example 1 did throws an error at pushing though, Example 2 didn't, but the compose fails.

**Comment by 7flying** [comment]
> Hello, I have been trying to test locally the firewall zones customization, may you please help me to clarify which is the proper syntax to include it in the blueprint?
> 
> Example 1: [customizations.firewall.zones] name = "dmz" sources = ["192.168.100.50"]
> 
> or
> 
> Example 2: [[customizations.firewall.zones]] name = "dmz" sources = ["192.168.100.50"]
> 
> Example 1 did throws an error at pushing though, Example 2 didn't, but the compose fails.

Example 2 is the way to go. You need to use an [array of tables](https://toml.io/en/v1.0.0#array-of-tables).
My tests have been done using osbuild at commit `1ecc7843866af7a1896f55ac0bba3482e7ad15d8` and osbuild-composer with this branch rebased in the current main, you need the commits from #3099 .

**Comment by achilleas-k** [comment]
CS9 CI runners are failing to install osbuild because the repo snapshots we have configured are too old.
We're going to have to update the repos in `test/data/repositories/centos-stream-9.json` to `20221101` from `20220330`.

**Comment by 7flying** [comment]
> CS9 CI runners are failing to install osbuild because the repo snapshots we have configured are too old. We're going to have to update the repos in `test/data/repositories/centos-stream-9.json` to `20221101` from `20220330`.

Can I do it myself with a new commit on this PR or do you want to handle that separately?

**Comment by achilleas-k** [comment]
> > CS9 CI runners are failing to install osbuild because the repo snapshots we have configured are too old. We're going to have to update the repos in `test/data/repositories/centos-stream-9.json` to `20221101` from `20220330`.
> 
> Can I do it myself with a new commit on this PR or do you want to handle that separately?

Go ahead and update them.  If we get any issues (from new packages), we can think about dealing with them separately.

File: test/data/repositories/centos-stream-9.json [hunk 1]
```diff
@@ -2,61 +2,61 @@
...
       "check_gpg": true
     },
```
Comment (7flying) [inline test/data/repositories/centos-stream-9.json:62]: note that I haven't updated `google-compute-engine` nor `google-cloud-sdk` below (just checking if that's OK)

Comment (achilleas-k) [reply test/data/repositories/centos-stream-9.json:62]: Yup.  All good.  We don't snapshot those.

**Review by achilleas-k — APPROVED** [review]

**Comment by 7flying** [comment]
Status as of 4/11:
Some of the tests in Gitlab's CI [fail](https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/3276139517) in CentOS-9  because there is a missing GPG key on `containers-common` ([bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=2123611)).

We need to wait until `containers-common-1-45.el9` appears in the composes, the Osbuild folks need to make a new snapshot and then we can update the repos at `test/data/repositories` with the proper snapshot.

**Comment by runcom** [comment]
@7flying can you rebase and update the repo to 20221115?

**Comment by 7flying** [comment]
@runcom, I will wait until #3136 goes in to rebase again

**Comment by runcom** [comment]
this needs a rebase :angel:

**Comment by 7flying** [comment]
> this needs a rebase angel

Yes I was working on it, but the whole firewall-customizations (https://github.com/osbuild/osbuild-composer/pull/3055/files#diff-35ef1e16afb78c169160d0547c67bcc94cb891f2519ee115dfcabd776bfd54d9R69) pre `new-shiny-rhel` are nowhere to be found, the stuff that was there before our changes.

**Comment by 7flying** [comment]
> > this needs a rebase angel
> 
> Yes I was working on it, but the whole firewall-customizations (https://github.com/osbuild/osbuild-composer/pull/3055/files#diff-35ef1e16afb78c169160d0547c67bcc94cb891f2519ee115dfcabd776bfd54d9R69) pre `new-shiny-rhel` are nowhere to be found, the stuff that was there before our changes.

Found the thing, ignore this.

**Review by achilleas-k — APPROVED** [review]
Thanks for this!

**Pull request merged**